crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -6717,6 +6717,29 @@ impl<'db> TypeInferenceBuilder<'db> {
         right: Type<'db>,
         range: TextRange,
     ) -> Result<Type<'db>, CompareUnsupportedError<'db>> {
+        let is_str_literal_in_tuple = |literal: Type<'db>, tuple: TupleType<'db>| {
```

Comment:
should we bail out here if it's a pathologically long tuple type, by checking the length of the tuple first?

Reply:
Do y'all have a sense of what "pathological long" means?

I would maybe start at `2^24`?

Reply:
That's fine with me!